server/src/main/java/org/opensearch/tracing/SpanHolder.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Mutable wrapper class to store {@link Span}.
+ */
+public class SpanHolder {
+
+    private final AtomicReference<Span> span = new AtomicReference<>();
+
+    public SpanHolder(SpanHolder spanHolder) {
+        if (spanHolder != null) {
+            this.span.set(spanHolder.span.get());
+        }
+    }
+
+    public SpanHolder(Span span) {
+        this.span.set(span);
+    }
+
+    public Span getSpan() {
+        return span.get();
+    }
+
+    public void setSpan(Span span) {
+        this.span.set(span);
+    }
```

Comment:
The structure looks inconsistent for instance the first one looks similar to a copy ctor for `SpanHolder ` then you have another ctor and a setter to achieve the same thing. It poorly mixes ctor and setters.
See https://github.com/opensearch-project/OpenSearch/pull/7648/files#r1209471404

Reply:
I had thought to have overloaded constructors will offload some extra manipulations of its users. But I agree it may look inconsistent, will remove the copy constructor.

For the second point regarding directly using Atomic reference, I think having a wrapper class improves readability. Otherwise either approach is Ok.

Reply:
This code is removed no